lib/router-config/src/traffic_shaping.rs

```diff
@@ -21,7 +21,10 @@ pub struct TrafficShapingExecutorConfig {
     /// A list of headers that should be used to fingerprint requests for deduplication.
     ///
     /// If not provided, the default is to use the "authorization" header only.
-    #[serde(default = "default_dedupe_fingerprint_headers")]
+    #[serde(
+        default = "default_dedupe_fingerprint_headers",
+        deserialize_with = "deserialize_and_normalize_dedupe_fingerprint_headers"
+    )]
     pub dedupe_fingerprint_headers: Vec<String>,
```

Comment:
nit: this could already be `HeaderName` instead of `String`, and then you can deserialize it (and validate that the string is an actual valid header name) while parsing